src/model.py

```diff
@@ -151,6 +153,16 @@ def init_engine(self):
             AsyncEngineArgs(**self.vllm_engine_config)
         )
 
+        # If TRITON_ENABLE_METRICS<_CPU/_GPU> build flag is enabled.
+        if self.args["metrics_mode"] in ["all", "cpu", "gpu"]:
```

Comment:
I'm curious if instead of adding a new `config` value, can we re-use any of vllm's arguments as a proxy?

Reply:
{
    "model_config": '{
        "name": "vllm_model",
        "platform": "",
        "backend": "vllm",
        "runtime": "model.py",
        "version_policy": {"latest": {"num_versions": 1}},
        "max_batch_size": 0,
        "input": [
            {
                "name": "text_input",
                "data_type": "TYPE_STRING",
                "format": "FORMAT_NONE",
                "dims": [1],
                "is_shape_tensor": false,
                "allow_ragged_batch": false,
                "optional": false,
                "is_non_linear_format_io": false,
            },
            {
                "name": "stream",
                "data_type": "TYPE_BOOL",
                "format": "FORMAT_NONE",
                "dims": [1],
                "is_shape_tensor": false,
                "allow_ragged_batch": false,
                "optional": true,
                "is_non_linear_format_io": false,
            },
            {
                "name": "sampling_parameters",
                "data_type": "TYPE_STRING",
                "format": "FORMAT_NONE",
                "dims": [1],
                "is_shape_tensor": false,
                "allow_ragged_batch": false,
                "optional": true,
                "is_non_linear_format_io": false,
            },
            {
                "name": "exclude_input_in_output",
                "data_type": "TYPE_BOOL",
                "format": "FORMAT_NONE",
                "dims": [1],
                "is_shape_tensor": false,
                "allow_ragged_batch": false,
                "optional": true,
                "is_non_linear_format_io": false,
            },
        ],
        "output": [
            {
                "name": "text_output",
                "data_type": "TYPE_STRING",
                "dims": [-1],
                "label_filename": "",
                "is_shape_tensor": false,
                "is_non_linear_format_io": false,
            }
        ],
        "batch_input": [],
        "batch_output": [],
        "optimization": {
            "priority": "PRIORITY_DEFAULT",
            "input_pinned_memory": {"enable": true},
            "output_pinned_memory": {"enable": true},
            "gather_kernel_buffer_threshold": 0,
            "eager_batching": false,
        },
        "instance_group": [
            {
                "name": "vllm_model_0",
                "kind": "KIND_MODEL",
                "count": 1,
                "gpus": [],
                "secondary_devices": [],
                "profile": [],
                "passive": false,
                "host_policy": "",
            }
        ],
        "default_model_filename": "",
        "cc_model_filenames": {},
        "metric_tags": {},
        "parameters": {},
        "model_warmup": [],
        "model_transaction_policy": {"decoupled": true},
    },
    "model_instance_kind": "MODEL",
    "model_instance_name": "vllm_model_0_0",
    "model_instance_device_id": "0",
    "model_repository": "/mnt/backend_repos/vllm_backend/samples/model_repository/vllm_model",
    "model_version": "1",
    "model_name": "vllm_model",
    "metrics_mode": "off",
}

Reply:
These are Triton arguments, I mentioned vllm engine arguments: https://docs.vllm.ai/en/latest/models/engine_args.html .

Reply:
Speculating here, so needs to be verified. Based on quick look, I think `disable_log_stats` should control whether to collect stats or not, based on this: https://github.com/vllm-project/vllm/blob/21b9c49aa37c7ba08590a99b0d4f15f86439c8f9/vllm/engine/llm_engine.py#L450  and this: https://github.com/vllm-project/vllm/blob/21b9c49aa37c7ba08590a99b0d4f15f86439c8f9/vllm/engine/llm_engine.py#L319C17-L319C26

So I think we can read `disable_log_stats` from `model.json` and enable metrics based on the presence of this flag. CC @rmccorm4 ad @GuanLuo

Reply:
In any outcome of this discussion, we should also test the behavior of our backend when `disable_log_stats` is present in `model.json` and when it's not.

Reply:
Resolved